### DISCUSSION TITLE
Dynamic Handles & Auto-Deleting Edges?

### DISCUSSION THREAD

**Discussion by noanflaherty** [xyflow/xyflow#3123] [Q&A] [answered]
Hello, my usage of react-flow heavily relies on Dynamic Handles for a given Node type. Various interactions may cause a Node to have more or fewer Handles over the course of its existence.

Everything works fine except for when an existing Handle with a connected Edge is removed. When this happens, the Edge remains and is still attached to the Node, and I get the following warning:

```
[React Flow]: Couldn't create edge for source handle id: "<id>", edge id: <id>. Help: https://reactflow.dev/error#008 
````

Instead, I'd rather the Edge be deleted altogether when either its Source or Target handles are dynamically removed.

Despite substantial digging through docs/source code and experimentation, I can't quite achieve this. Is this possible?

**Comment by bcakmakoglu** [comment, marked as answer]
You could just remove the edge using [`deleteElements`](https://github.com/Aigoritma/automl-frontend/pull/944)

```ts
const { deleteElements } = useReactFlow()

const someCallback = () => {
	deleteElements({ edges: [{ id: 'some-edge-id' }] })
}
```

**Reply by noanflaherty** [reply]
This does the trick, thanks!

In case it helps others, I didn’t have an explicit callback, so had to rely on a useEffect to listen for changes that would result in the number of handles changing, then explicitly remove edges from within.
